bug(workflow): worker [ERROR] silently skips lead review — errors swallowed

## Problem

In \`src/lib/workflow.ts:499-504\`, the \`workersConverged\` check includes \`'[error]'\` as a convergence signal:

\`\`\`ts
const workersConverged = workerTurns.length > 0 && workerTurns.every(t => {
  const lower = t.content.toLowerCase();
  return lower.includes('nothing to do') || lower.includes('pr created') ||
    lower.includes('issue closed') || lower.includes('all tasks complete') ||
    lower.includes('no open issues') || lower.includes('[error]');  // ← BUG
});
\`\`\`

When a worker fails (produces `[ERROR] Agent X failed: ...`), `workersConverged` is `true`, so the lead review turn is **skipped entirely**. The lead never sees the error, cannot retry or escalate, and the cycle continues as if work succeeded.

## Solution

Remove `'[error]'` from `workersConverged` conditions. Worker errors should always trigger a lead review turn so the lead can diagnose, retry, or escalate.

\`\`\`ts
return lower.includes('nothing to do') || lower.includes('pr created') ||
  lower.includes('issue closed') || lower.includes('all tasks complete') ||
  lower.includes('no open issues');
\`\`\`

## Files
- \`src/lib/workflow.ts:499-504\` — remove \`[error]\` from workersConverged check

## Testing
- [ ] \`npm run build\` passes
- [ ] When a worker turn contains \`[ERROR]\`, lead review still executes
- [ ] Unit test: \`workersConverged\` returns false for error-only worker output

## Labels
type:code, priority:P1, squad:cli

--- a/src/lib/conversation.ts
+++ b/src/lib/conversation.ts
@@ -127,6 +127,19 @@ const CONVERGENCE_PHRASES = [
   'no open issues', 'no pending tasks', 'no pending issues',
 ];
 
+/** Phrases in a verifier turn that signal approval → converge */
+const VERIFIER_APPROVAL_PHRASES = [
+  'approved', 'lgtm', 'looks good',
+  'all checks pass', 'all tests pass', 'tests pass',
+  'passed', 'quality standards met',
+];
+
+/** Phrases in a verifier turn that signal rejection → continue cycle */
+const VERIFIER_REJECTION_PHRASES = [
+  'failed', 'rejected', 'needs fixes', 'needs changes',
+  'does not pass', 'did not pass', 'failing',
+];
+
 /** Phrases that indicate more work needed */
 const CONTINUATION_PHRASES = [
   'needs review', 'needs feedback', 'needs input', 'need clarification',
@@ -166,9 +179,21 @@ export function detectConvergence(
 
   const lastTurn = transcript.turns[transcript.turns.length - 1];
   const content = lastTurn.content;
+  const lower = content.toLowerCase();
+
+  // Verifier turns: check approval/rejection before generic signals
+  if (lastTurn.role === 'verifier') {
+    const rejected = VERIFIER_REJECTION_PHRASES.some(phrase => lower.includes(phrase));
+    if (rejected) {
+      return { converged: false, reason: 'Verifier rejected — continuing cycle' };
+    }
+    const approved = VERIFIER_APPROVAL_PHRASES.some(phrase => lower.includes(phrase));
+    if (approved) {
+      return { converged: true, reason: 'Verifier approved' };
+    }
+  }
 
   // Continuation signals beat convergence (bias toward completing work)
-  const lower = content.toLowerCase();
   const hasContinuation = CONTINUATION_PHRASES.some(phrase => lower.includes(phrase));
   if (hasContinuation) {
     return { converged: false, reason: 'Continuation signal detected' };

--- a/src/lib/workflow.ts
+++ b/src/lib/workflow.ts
@@ -27,7 +27,6 @@ import {
 import {
   type Squad,
   findSquadsDir,
-  loadAgentDefinition,
 } from './squad-parser.js';
 
 // =============================================================================
@@ -74,7 +73,6 @@ function executeAgentTurn(config: AgentTurnConfig): string {
   const { agentName, agentPath, role, squadName, model, transcript, task } = config;
 
   // Build the prompt: agent definition + transcript context + role instructions
-  const definition = loadAgentDefinition(agentPath);
   const transcriptContext = serializeTranscript(transcript);
 
   let roleInstructions: string;
@@ -411,6 +409,9 @@ export async function runConversation(
         transcript,
         cwd: squadCwd,
       });
+      if (output.startsWith('[ERROR]')) {
+        process.stderr.write(`  [WARN] Worker ${workers[0].name} errored: ${output}\n`);
+      }
       addTurn(transcript, workers[0].name, 'worker', output, estimateTurnCost(options.model || 'sonnet'));
     } else if (workers.length > 1) {
       log(`Turns ${transcript.turns.length + 1}-${transcript.turns.length + workers.length}: ${workers.map(w => w.name).join(', ')} (workers, parallel)`);
@@ -427,6 +428,9 @@ export async function runConversation(
       );
       const workerResults = await Promise.all(workerPromises);
       for (const { agent, output } of workerResults) {
+        if (output.startsWith('[ERROR]')) {
+          process.stderr.write(`  [WARN] Worker ${agent.name} errored: ${output}\n`);
+        }
         addTurn(transcript, agent.name, 'worker', output, estimateTurnCost(options.model || 'sonnet'));
       }
     }

